docs/source/reference/comparison.rst

```diff
@@ -0,0 +1,178 @@
+
+Here is a side-by-side comparison of the YAMLs for serving Gemma with vLLM on SkyPilot vs Kubernetes:
+
+.. raw:: html
+
+   <div class="row">
+       <div class="col-md-6 mb-3">
+            <h3> SkyPilot (19 lines) </h3>
+
+.. code-block:: yaml
+   :linenos:
+
+   envs:
+     MODEL_NAME: google/gemma-2b-it
+     HF_TOKEN: myhftoken
+
+   resources:
+     image_id: docker:vllm/vllm-openai:latest
+     accelerators: L4:1
+     ports: 8000
+
+   setup: |
+     conda deactivate
+     python3 -c "import huggingface_hub; huggingface_hub.login('${HF_TOKEN}')"
+
+   run: |
+     conda deactivate
+     echo 'Starting vllm openai api server...'
+     python -m vllm.entrypoints.openai.api_server \
+     --model $MODEL_NAME --tokenizer hf-internal-testing/llama-tokenizer \
+     --host 0.0.0.0
+
+.. raw:: html
+
+       </div>
+       <div class="col-md-6 mb-3">
+            <h3> Kubernetes (65 lines) </h3>
+
+.. code-block:: yaml
+   :linenos:
+
+   apiVersion: apps/v1
+   kind: Deployment
+   metadata:
+     name: vllm-gemma-deployment
+   spec:
+     replicas: 1
+     selector:
+       matchLabels:
+         app: gemma-server
+     template:
+       metadata:
+         labels:
+           app: gemma-server
+           ai.gke.io/model: gemma-1.1-2b-it
+           ai.gke.io/inference-server: vllm
+           examples.ai.gke.io/source: user-guide
+       spec:
+         containers:
+         - name: inference-server
+           image: us-docker.pkg.dev/vertex-ai/ vertex-vision-model-garden-dockers/pytorch-vllm-serve:20240527_0916_RC00
+           resources:
+             requests:
+               cpu: "2"
+               memory: "10Gi"
+               ephemeral-storage: "10Gi"
+               nvidia.com/gpu: 1
+             limits:
+               cpu: "2"
+               memory: "10Gi"
+               ephemeral-storage: "10Gi"
+               nvidia.com/gpu: 1
+           command: ["python3", "-m", "vllm.entrypoints.api_server"]
+           args:
+           - --model=$(MODEL_ID)
+           - --tensor-parallel-size=1
+           env:
+           - name: MODEL_ID
+             value: google/gemma-1.1-2b-it
+           - name: HUGGING_FACE_HUB_TOKEN
+             valueFrom:
+               secretKeyRef:
+                 name: hf-secret
+                 key: hf_api_token
+           volumeMounts:
+           - mountPath: /dev/shm
+             name: dshm
+         volumes:
+         - name: dshm
+           emptyDir:
+             medium: Memory
+         nodeSelector:
+           cloud.google.com/gke-accelerator: nvidia-l4
+   ---
+   apiVersion: v1
+   kind: Service
+   metadata:
+     name: llm-service
+   spec:
+     selector:
+       app: gemma-server
+     type: ClusterIP
+     ports:
+       - protocol: TCP
+         port: 8000
+         targetPort: 8000
+
+.. raw:: html
+
+       </div>
+   </div>
+
```

Comment:
Just thinking out loud, not sure if we can think of other value adds on this page, or if we can go a bit deeper for the points we mentioned below, like what is the quantitative benefits `GPU availability` and `Costs` can provide. The current points feel a bit weak.

Reply:
Hmm we had some numbers in the earlier commits but removed them in [this comment](https://github.com/skypilot-org/skypilot/pull/3756#discussion_r1683717038)

Reply:
I see! Let's skip the numbers for now then.